sys/net/skald/skald.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_skald
+ * @{
+ *
+ * @file
+ * @brief       Skald's link layer implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "assert.h"
+#include "random.h"
+#include "luid.h"
+
+#include "net/netdev/ble.h"
+#include "net/skald.h"
+
+#define ENABLE_DEBUG            (0)
+#include "debug.h"
+
+#define JITTER_MIN              (0U)            /* 0ms */
+#define JITTER_MAX              (10000U)        /* 10ms */
+
+#define ADV_CHAN_NUMOF          sizeof(_adv_chan)
+#define ADV_AA                  (0x8e89bed6)    /* access address */
+#define ADV_CRC                 (0x00555555)    /* CRC initializer */
+
+static const uint8_t _adv_chan[] = SKALD_ADV_CHAN;
+
+static netdev_ble_ctx_t _ble_ctx = {
+    .aa.u32 = ADV_AA,
+    .crc = ADV_CRC,
+};
+
+static netdev_t *_radio;
+
+static void _stop_radio(void)
+{
+    netdev_ble_stop(_radio);
+    _radio->context = NULL;
+}
+
+static void _sched_next(skald_ctx_t *ctx)
+{
+    ctx->last += SKALD_INTERVAL;
+    /* schedule next advertising event, adding a random jitter between
+     * 0ms and 10ms (see spec v5.0-vol6-b-4.4.2.2.1) */
+    ctx->last += random_uint32_range(JITTER_MIN, JITTER_MAX);
+    /* compensate the time passed since the timer triggered last by using the
+     * current value of the timer */
+    xtimer_set(&ctx->timer, (ctx->last - xtimer_now_usec()));
+}
+
+static void _on_adv_evt(void *arg)
+{
+    skald_ctx_t *ctx = (skald_ctx_t *)arg;
+
+    /* advertise on the next adv channel - or skip this event if the radio is
+     * busy */
+    if (_radio->context != NULL) {
+
```

Comment:
Is there something missing here? Otherwise: the if is redundant.

Reply:
fixed.